sympy/matrices/expressions/matmul.py

```diff
@@ -175,6 +175,27 @@ def args_cnc(self, **kwargs):
         coeff_nc = [x for x in self.args if not x.is_commutative]
         return [coeff_c, coeff_nc]
 
+    # XXX: This is needed because MatPow doesn't inherit from Pow so powers
+    # aren't expanded to multiplications during matching. Maybe there's a
+    # better way to do this?
```

Comment:
I'm not a fan of this hack. Can we just make it subclass from Pow? We already have MatMul and MatAdd subclassing from Mul and Add.